backend/endpoints/trades/route.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Literal, Tuple
+from decimal import Decimal
+
+from ninja import Schema, Router
+from django.utils import timezone
+from django.db import transaction
+
+from models.models import ArticleModel, HoldingModel, StockModel, TradeModel
+from models.models import UserModel
+from endpoints.auth import AuthBearer, AuthenticatedRequest
+from endpoints.error import FriendlyClientException
+from lib.polygon_api import PolygonAPI
+
+
+router = Router(auth=AuthBearer())
+
+
+class TradeSchema(Schema):
+    ticker: str
+    units_change: float
+    balance_change: float
+    time: int
+    text_evidence: str
+    article_evidence: List[str]
+
+    @staticmethod
+    def from_model(model: TradeModel) -> 'TradeSchema':
+        return TradeSchema(
+            ticker=model.stock.ticker,
+            units_change=float(model.units_change),
+            balance_change=float(model.balance_change),
+            time=int(model.time.timestamp()),
+            text_evidence=model.text_evidence,
+            article_evidence=[article.article_id for article in model.article_evidence.all()])
+
+
+@router.get("/trades/personal", response=List[TradeSchema])
+def personal_trades(request: AuthenticatedRequest) -> List[TradeSchema]:
+    user = UserModel.objects.get(username=request.auth)
+    trades = TradeModel.objects.filter(user=user)
+
+    return [TradeSchema.from_model(trade) for trade in trades]
+
+
+class AddTradeSchema(Schema):
+    ticker: str
+    side: Literal['BUY'] | Literal['SELL']
+    type: Literal['UNITS'] | Literal['PRICE']
+    amount: float
+    text_evidence: str
+    article_evidence: List[str]
+
+    def balance_units_changes(self) -> Tuple[float, float]:
+        if self.amount <= 0:
+            raise FriendlyClientException(
+                "Trade amount must be greater than 0!")
+
+        current_price = PolygonAPI().recent_price(self.ticker)
+        deal_price = current_price.high if self.side == 'BUY' else current_price.low
```

Comment:
Maybe not something for right now, but I'm kinda worried about this producing inconsistent prices from that displayed on the UI since that would involve another API call at an earlier time. So maybe consider receiving `deal_price` from the frontend instead? (Not the best solution but since it's only simulated trading...)

Reply:
Dealing with inconsistent prices is a feature of basically every trading platform - in my experience platforms very rarely quote a price that you have time to lock in, and usually orders are filled at next market price.

Reply:
Then it shouldn't be a problem :)))